azurerm/internal/services/logic/parse/logic_app_workflow_test.go

```diff
@@ -0,0 +1,69 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestLogicAppWorkflowID(t *testing.T) {
+	testData := []struct {
+		Name   string
+		Input  string
+		Error  bool
+		Expect *LogicAppWorkflowId
+	}{
+		{
+			Name:  "Empty",
+			Input: "",
+			Error: true,
+		},
+		{
+			Name:  "No Resource Groups Segment",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000",
+			Error: true,
+		},
+		{
+			Name:  "No Resource Groups Value",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/",
+			Error: true,
+		},
+		{
+			Name:  "No Workflow Name",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/workflows",
+			Error: true,
+		},
+		{
+			Name:  "Correct case",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/workflows/workflow1",
+			Expect: &LogicAppWorkflowId{
+				Subscription:  "00000000-0000-0000-0000-000000000000",
```

Comment:
can we update the test data to use a non-default value here? for now we can likely remove this in favour of the ID Formatters